evaluator/builtin_time.go

```diff
@@ -673,6 +675,416 @@ func builtinFromUnixTime(args []types.Datum, _ context.Context) (d types.Datum,
 	return builtinDateFormat([]types.Datum{d, args[1]}, nil)
 }
 
+// strToDate convert date string according to format, return true on success,
```

Comment:
builtinDateFormat would return a NULL datum if input date and format don't match, so we don't need return error